### DISCUSSION TITLE
Clear App Data programmatically

### DISCUSSION THREAD

**Discussion by majco333** [cypress-io/cypress#24667] [Questions and Help]
Is there any way to call clear App Data? I know, I can do it like this: https://docs.cypress.io/guides/references/troubleshooting#To-clear-App-Data
But I need it in CI build machine automatically (programmatically)

**Comment by portyu9** [comment]
Hi @majco333 

There isn't a dedicated `cypress clear-app-data` CLI command.

Also, `cypress cache clear` is **not** the equivalent of **Clear App Data** — it deletes cached Cypress binary installations.

For CI, the best solution is normally to use a disposable/clean worker for every run. If you specifically need to automate the same cleanup, current Cypress documentation lists the App Data locations as:

- macOS: `~/Library/Application Support/Cypress`
- Linux: `~/.config/Cypress`
- Windows: `%APPDATA%\Cypress`

So on a disposable Linux CI agent, for example, you can remove the Cypress App Data directory before starting Cypress:

```bash
rm -rf ~/.config/Cypress
```

Do this before Cypress is running, not while the Cypress process is using those files.

If the goal is merely to guarantee clean browser state between tests, don't delete App Data for that — Cypress test isolation handles cookies, local storage and session storage separately.

Please mark this answer as accepted if it helped, thank you.
